kinds/structured/preferences/preferences_kind.cue

```diff
@@ -0,0 +1,49 @@
+package kind
+
+name: "Preferences"
+maturity: "merged"
+
+lineage: seqs: [
+	{
+		schemas: [
+			{//0.0
+				// UID for the home dashboard
+				homeDashboardUID?: string
```

Comment:
@sdboyer -- what is the best pattern for having optional strings that are not pointers in go? can this be a string with an `json:“homeDashboardUID,omitempty”`?

Reply:
some discussion here https://github.com/grafana/grok/issues/1

i think that there are some cases where empty string and absence being equivalent may make sense. And if the CUE type doesn't otherwise allow `""`, then we could probably generate without a pointer, because it wouldn't introduce an ambiguity.

But in general, pointers are the only unambiguous way to express optionality in Go. The simple answer for now is to expect callers to rely on a `Ptr()` func, as outlined in that issue.

(We're starting to build up a backlog of codegen questions/issues and this one does belong on it - cc @rhorvath)

Reply:
to be clear, the ergonomics here do irk the shit out of me and i want a solution...it's just a warren of things that aren't the highest impact to think about rn

but, to that end -

> if the CUE type doesn't otherwise allow `""`

simplest way of saying this in the general case is like this:

```cue
field?: string & strings.MinRunes(1)
```

this won't produce the pointerless Go you want right now, but it is the simplest thing i can immediately think of for precisely specifying rather than falling back on ambiguity

Reply:
makes sense -- just wanted to make sure I am not missing something basic.

As long as we are comfortable adding this constraint when our generation would be nicer to work with, it is not urgent

Reply:
to be clear i'm not _promising_ a more ergonomic outcome here, because i don't know what it would be and i don't have a strong grasp on all the relevant constraints.

i know how i intend to address this problem in grok, but idk about its applicability here